src/AppBundle/GitHooks/Action/CheckLockFileCommitted.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+declare(strict_types=1);
+
+namespace AppBundle\GitHooks\Action;
+
+use CaptainHook\App\Config;
+use CaptainHook\App\Console\IO;
+use SebastianFeldmann\Git\Repository;
+
+final class CheckLockFileCommitted extends Action
+{
+    protected const ERROR_MESSAGE = 'There is no composer.lock file in your commit!';
+
+    public function execute(Config $config, IO $io, Repository $repository, Config\Action $action): void
+    {
+        if (!$this->isEnabled($action)) {
+            return;
+        }
+
+        $changedJsonFiles = $repository->getIndexOperator()->getStagedFilesOfType('json');
+
+        if (!\in_array('composer.json', $changedJsonFiles, true)) {
+            $io->write('<info>No composer.json detected in commit.</info>');
```

Comment:
Can we remove the output if action does nothing to reduce the load on users when there's nothing to do anyway?

Reply:
Done in af51ba57